lib/web_ui/dev/integration_tests_manager.dart

```diff
@@ -47,18 +47,17 @@ class IntegrationTestsManager {
   }
 
   Future<bool> _runPubGet(String workingDirectory) async {
-    final String executable = isCirrus ? environment.pubExecutable : 'flutter';
-    final List<String> arguments = isCirrus
-        ? <String>[
-            'get',
-          ]
-        : <String>[
-            'pub',
-            'get',
-          ];
+    if (isCi) {
```

Comment:
By default we should always clone the Flutter repo. Otherwise we won't be able to reproduce breakages locally due to the different versions of the Flutter framework between CI and the local checkout.

However, if someone needs a specific version of Flutter that's different from what `_cloneFlutterRepo()` uses, we can provide a `--local-framework` option (similar to `--local-engine` that the `flutter` command uses when you want a different version of the engine from the one that's fetched by default).

Reply:
I do not agree. Actually we already discussed this before and made an agreement. We even have it as a task in the issue :)

I think (we also agreed back in the day) for a local developer felt test should not download the flutter repo. At least for now, in this age of internet issues. If a developer uses what they already get used:

```
felt test
```
then if it clones all the flutter repo without asking them, this will be a very bad experience.

```
felt test --always-clone-flutter 
```

can be used for a local developer for downloading flutter.  Felt test will always download it in CIs.

Reply:
> I do not agree. Actually we already discussed this before and made an agreement. We even have it as a task in the issue :)

My understanding is we do not want to clone Flutter when running engine unit-tests. Those tests do not need the Flutter framework and so it makes sense to skip it. However, integration tests do need it, and for correct results they need a specific version of it, not a random version that happens to be in `PATH`.

When running `felt test` we already download a bunch of stuff:

* We clone flutter/goldens (in the future we'll be downloading from Skia Gold).
* We download Chromium or Firefox when running unit-tests.

So I think the ship is sailed in terms off internet issues. Since this only affects Flutter contributors, it's not a big issue. However, correctness is a big issue. I think the network issue can be alleviated by caching the Flutter checkout between tests, just like we cache Chromium. Even when the cached version is old, we can do `git pull` instead of `git clone` so we only fetch the delta as opposed to the entire repo.

Let's not optimize at the expense of correctness. It will bite us later. We can deal with the network issue when and if it arises.

Reply:
Thanks for the suggestion. Here are the changes I made:
- I added a new flag: use-system-flutter (false by default)
- unless use-system-flutter is set, I always clone the repo.
- I clean the repo when felt clean is used.
- I always clean the repo before cloning a new one.

I added a todo. Later I'll also implement the git pull suggestion.